py/configu/commands/eval_command.py

```diff
@@ -158,24 +158,36 @@ def _validate_result(self, result: EvalCommandReturn):
             for key, value in result.items():
                 cfgu = value["context"]["cfgu"]
                 evaluated_value = value["result"]["value"]
-                type_test = ConfigSchema.CFGU.VALIDATORS.get(
-                    cfgu.type.value, lambda: False
-                )
-                test_values = (
-                    (
-                        evaluated_value,
-                        cfgu.pattern,
+                try:
+                    type_test = ConfigSchema.CFGU.VALIDATORS[cfgu.type.value]
+                    test_values = (
+                        (
+                            evaluated_value,
+                            cfgu.pattern,
+                        )
+                        if cfgu.type == CfguType.REG_EX
+                        else (evaluated_value,)
                     )
-                    if cfgu.type == CfguType.REG_EX
-                    else (evaluated_value,)
-                )
-                if not type_test(*test_values):
-                    raise ValueError(
+                    if not type_test(*test_values):
+                        raise ValueError(
+                            error_message(
+                                f"invalid value type for key '{key}'", error_scope
+                            ),
+                            f"value '{test_values[0]}' must be a " f"'{cfgu.type}'",
+                        )
+                except KeyError:
```

Comment:
I'd recommend wrapping only line 162 in the `try-except`. That way it's clear what caused the error to happen.
For example, if in future a poorly constructed `type_test` ends up raising a `KeyError` we end up raising a misleading error message - the particular `CfguType` is _technically_ supported, however there is a bug in the particular implementation.

Reply:
Agreed. 
Fixed.
Thanks!